athena/combine_coverage.py

```diff
@@ -0,0 +1,257 @@
+            class_elem.set('complexity', str(cls['complexity']))
+            
+            # Add methods element (required by Cobertura format)
+            methods_elem = ET.SubElement(class_elem, 'methods')
+            
+            # Add lines element with detailed line information
+            lines_elem = ET.SubElement(class_elem, 'lines')
+            for line in cls.get('lines', []):
+                line_elem = ET.SubElement(lines_elem, 'line')
+                line_elem.set('number', str(line['number']))
+                line_elem.set('hits', str(line['hits']))
+            
+            # Add branches element with detailed branch information
+            if cls.get('branches'):
+                branches_elem = ET.SubElement(class_elem, 'branches')
+                for branch in cls['branches']:
+                    branch_elem = ET.SubElement(branches_elem, 'branch')
+                    branch_elem.set('number', str(branch['number']))
+                    branch_elem.set('hits', str(branch['hits']))
+    
+    # Write the XML file
+    tree = ET.ElementTree(root)
+    tree.write(output_file, encoding='utf-8', xml_declaration=True)
+
+def main():
+    """Main function to combine coverage files."""
+    if len(sys.argv) < 3:
+        print("Usage: python combine_coverage.py [--summary] <output_file> <input_file1> [input_file2] ...")
+        sys.exit(1)
+    
+    # Check for summary flag
+    show_summary = False
+    if sys.argv[1] == "--summary":
+        show_summary = True
+        sys.argv = sys.argv[1:]  # Remove the flag
+    
+    output_file = sys.argv[1]
+    input_files = sys.argv[2:]
+    
+    print(f"Combining {len(input_files)} coverage files...")
+    
+    # Parse all input files
+    all_packages = []
+    for xml_file in input_files:
+        if os.path.exists(xml_file):
+            print(f"  Parsing {xml_file}...")
+            packages = parse_coverage_xml(xml_file)
+            all_packages.append(packages)
+        else:
+            print(f"  Warning: {xml_file} not found, skipping...")
+    
+    if not all_packages:
+        print("No valid coverage files found!")
+        sys.exit(1)
+    
+    # Combine packages
+    print("Combining duplicate packages...")
+    combined = combine_packages(all_packages)
+    
+    # Create output file
+    print(f"Creating combined coverage file: {output_file}")
+    create_combined_xml(combined, output_file)
+    
+    print(f"✅ Combined coverage saved to {output_file}")
+    print(f"📊 Combined {len(combined)} unique packages")
+    
+    # Show summary if requested
+    if show_summary:
+        print("\n📊 Combined Coverage Summary:")
+        print("=" * 80)
+        print(f"{'Package':<60} {'Line Rate':<10} {'Branch Rate':<12} {'Status'}")
+        print("=" * 80)
+        total_lines = 0
+        total_covered = 0
+        total_branches = 0
+        total_branch_covered = 0
+        
+        for name, data in sorted(combined.items()):
```

Comment:
Instead of the plain text, can we also generate a markdown table here?
<img width="993" height="697" alt="image" src="https://github.com/user-attachments/assets/1ab475ce-2937-4e69-8d46-4cf36555e727" />

So that it basically looks like the output from the irongut/CodeCoverageSummary@v1.3.0 action?